### DISCUSSION TITLE
How to match another rule first

### DISCUSSION THREAD

**Discussion by cathaysia** [tree-sitter/tree-sitter#3273] [Q&A] [answered]
I have the following text: 

```
//@identity
// something 
```
the first is a annotaion, which always starts with `//@` then following a identity, the second is a comment, and both of them will be a `extras` member.

when there no `$.comment` in extras, tree-sitter works well, and it can match annotation normally, but when `$.comment` in extras, tree-sitter always match comment, how can I let tree-sitter match annotaion first, then match comment if failed?

here is my rules:

```js

anno: $ => seq('//@', $.identity),
comment: _ => seq('//', /(\\+(.|\r?\n)|[^\\\n])*/),
```

**Comment by ObserverOfTime** [comment]
Try using [`prec()`](https://tree-sitter.github.io/tree-sitter/creating-parsers#using-precedence)

**Comment by cathaysia** [comment, marked as answer]
this issue are solved by https://github.com/nvim-treesitter/nvim-treesitter/pull/6491#discussion_r1581079749
